asv_bench/benchmarks/gil.py

```diff
@@ -320,3 +325,43 @@ def time_nogil_kth_smallest(self):
         def run(arr):
             algos.kth_smallest(arr, self.k)
         run()
+
+
+class nogil_read_csv(object):
+    number = 1
+    repeat = 5
+
+    def setup(self):
+        if (not have_real_test_parallel):
+            raise NotImplementedError
+        # Using the values
+        self.df = DataFrame(np.random.randn(10000, 10))
+        self.df.to_csv('__test__.csv')
+
+        self.rng = date_range('1/1/2000', periods=10000)
+        self.df_date_time = DataFrame(np.random.randn(10000, 10), index=self.rng)
+        self.df_date_time.to_csv('__test_datetime__.csv')
+
+        self.df_object = DataFrame('foo', index=self.df.index, columns=self.create_cols('object'))
+        self.df_object.to_csv('__test_object__.csv')
+
+    def create_cols(self, name):
+        return [('%s%03d' % (name, i)) for i in range(5)]
+
+    def time_nogil_read_csv(self):
+        @test_parallel(num_threads=2)
+        def run(arr):
+            read_csv('__test__.csv', sep=',', header=None, float_precision=None)
```

Comment:
is their a cleanup function on these? (e.g. to remove the created csvs)?

@qwhelan ?